flink-runtime/src/test/java/org/apache/flink/runtime/state/StateBackendTestBase.java

```diff
@@ -2913,7 +2913,21 @@ public void testMapState() throws Exception {
 		assertEquals(new HashMap<Integer, String>() {{ put(103, "103"); put(1031, "1031"); put(1032, "1032"); }},
 				getSerializedMap(restoredKvState2, "3", keySerializer, VoidNamespace.INSTANCE, namespaceSerializer, userKeySerializer, userValueSerializer));
 
-		backend.dispose();
+		// [FLINK-10267] validate arbitrary iterator access not throwing IllegalStateException
```

Comment:
The test cases are already quite huge, can we create a separate one for this behaviour?